subprojects/core/src/main/java/org/gradle/internal/classpath/transforms/InstrumentingClassTransform.java

```diff
@@ -497,15 +284,51 @@ private static String getBoostrapMethodName(BytecodeInterceptorFilter intercepto
             }
         }
 
-        private String binaryClassNameOf(String className) {
-            return getObjectType(className).getClassName();
-        }
-
         private boolean isGroovyIndyCallsite(Handle bootstrapMethodHandle) {
             return (bootstrapMethodHandle.getOwner().equals(GROOVY_INDY_INTERFACE_TYPE) ||
                 bootstrapMethodHandle.getOwner().equals(GROOVY_INDY_INTERFACE_V7_TYPE)) &&
                 bootstrapMethodHandle.getName().equals("bootstrap") &&
                 bootstrapMethodHandle.getDesc().equals(GROOVY_INDY_INTERFACE_BOOTSTRAP_METHOD_DESCRIPTOR);
         }
+
+        private Object maybeInstrumentBootstrapArgument(Object argument) {
+            if (argument instanceof Handle) {
+                return maybeInstrumentHandle((Handle) argument);
+            }
+            return argument;
+        }
+
+        private Handle maybeInstrumentHandle(Handle handle) {
+            BridgeMethod bridgeMethod = owner.bridgeMethods.computeIfAbsent(handle, this::maybeBuildBridgeMethod);
+            if (bridgeMethod != null) {
+                return bridgeMethod.bridgeMethodHandle;
+            }
+            return handle; // No instrumentation requested.
+        }
+
+        @Nullable
+        private BridgeMethod maybeBuildBridgeMethod(Handle handle) {
+            for (JvmBytecodeCallInterceptor interceptor : interceptors) {
+                BridgeMethodBuilder methodBuilder = interceptor.findBridgeMethodBuilder(className, handle.getTag(), handle.getOwner(), handle.getName(), handle.getDesc());
+                if (methodBuilder != null) {
+                    return new BridgeMethod(makeBridgeMethodHandle(makeBridgeMethodName(handle), methodBuilder.getBridgeMethodDescriptor()), methodBuilder);
+                }
+            }
+            return null;
+        }
+
+        private String makeBridgeMethodName(Handle originalHandle) {
+            int index = owner.bridgeMethods.size();
```

Comment:
❓ Why is here using `index = size()` always safe?

Reply:
>Why is here using index = size() always safe?

I made it more obvious with an explicit counter.